rasa/nlu/utils/spacy_utils.py

```diff
@@ -90,10 +84,57 @@ def cache_key(
 
         # Fallback, use the language name, e.g. "en",
         # as the model name if no explicit name is defined
-        spacy_model_name = component_meta.get("model", model_metadata.language)
+        spacy_model_name = cls.check_model_fallback(
+            component_meta.get("model"), model_metadata.language, warn=True
+        )
 
         return cls.name + "-" + spacy_model_name
 
+    @classmethod
+    def check_model_fallback(cls, spacy_model_name, language_name, warn=False):
+        """
+        This method checks if the `spacy_model_name` is missing and attempts to do a fallback.
+        This feature is a measure to support spaCy 3.0 without breaking on users. In the future
+        spaCy will no longer support `spacy link`.
+        """
+        if not spacy_model_name:
+            if warn:
+                logger.warning(
```

Comment:
should be a `warnings.warn` as it's user facing (see https://www.notion.so/rasa/Python-Code-Conventions-26835ffd72484dd699f28fa693916d2e)